WelsonJS.Toolkit/WelsonJS.Launcher/ResourceServer.cs

```diff
@@ -430,24 +431,29 @@
             }
         }
 
-        private async void FetchBlobConfig()
+        private async Task FetchBlobConfig()
         {
             try
             {
                 string url = Program.GetAppConfig("BlobConfigUrl");
-                var response = await _httpClient.GetStreamAsync(url);
+                if (string.IsNullOrWhiteSpace(url))
+                {
+                    _logger?.Warn("BlobConfigUrl is not configured.");
+                    return;
+                }
 
-                var serializer = new XmlSerializer(typeof(BlobConfig));
+                using (var response = await _httpClient.GetStreamAsync(url))
                 using (var reader = new StreamReader(response))
                 {
-                    _blobConfig = (BlobConfig)serializer.Deserialize(reader);
+                    var serializer = new XmlSerializer(typeof(BlobConfig));
+                     _blobConfig = (BlobConfig)serializer.Deserialize(reader);
                 }
 
                 _blobConfig?.Compile();
             }
             catch (Exception ex)
             {
-                _logger?.Error($"Failed to fetch a blob config. Exception: {ex.Message}");
+                _logger?.Error($"Failed to fetch a blob config. Exception: {ex}");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Avoid string interpolation within log messages, as it can blur the lines between variables and the log content. Instead, opt for structured logging, where variables are passed as separate arguments. This improves the clarity of logs and facilitates efficient log analysis. For more information, refer to https://github.com/NLog/NLog/wiki/How-to-use-structured-logging.